metal/datasource_metal_vlan_test.go

```diff
@@ -97,6 +110,16 @@ func TestAccMetalDatasourceVlan_ByVxlanMetro(t *testing.T) {
 						"metal_vlan.foovlan", "id",
 						"data.metal_vlan.dsvlan", "id",
 					),
+					resource.TestCheckResourceAttr(
```

Comment:
I added the `barvlan` and related tests to replicate the example in the issue this PR aims to fix. This E2E test was passing, however. So I added unit tests to simulate the problems and I found and addressed it that way.

I can remove these added E2E changes if they seem unnecessary.